drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -134,7 +134,7 @@ static void nrf5_rx_thread(void *arg1, void *arg2, void *arg3)
 		LOG_DBG("Frame received");
 
 		pkt = net_pkt_rx_alloc_with_buffer(nrf5_radio->iface, pkt_len,
```

Comment:
Could you add a comment telling why we can wait forever here, I know it is in commit message but it helps if there is description here too.